### PR TITLE
feat: add support for openapi 3.1

### PR DESCRIPTION
**Pull request by pevisscher**

### CODE REVIEW

**Comment by pevisscher** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/skyleague/event-horizon/217?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#217** <a href="https://app.graphite.dev/github/pr/skyleague/event-horizon/217?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/skyleague/event-horizon/217?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#216** <a href="https://app.graphite.dev/github/pr/skyleague/event-horizon/216?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: src/spec/openapi/openapi.ts [hunk 1]
```diff
@@ -194,20 +262,73 @@ export function openapiFromHandlers(handlers: Record<string, unknown>, options:
...
+                trackSchema('schema' in r ? (r.schema as JsonSchema) : (r.body?.schema as JsonSchema))
+            }
+        }
+    }
+
+    // Set jsonSchemaDialect if schemas were found
+    // if (targetOpenapi === '3.1.0' && schemaFrequencies.size > 0) {
+    //     // biome-ignore lint/suspicious/noExplicitAny: <explanation>
+    //     ;(openapi as any).jsonSchemaDialect = Array.from(schemaFrequencies.entries()).reduce((a, b) => (b[1] > a[1] ? b : a))[0]
+    // }
```
Comment (MickVanDuijn) [inline src/spec/openapi/openapi.ts:283]: is this not needed? I'm still seeing the `schemaFrequencies` being defined, but never used

**Review by MickVanDuijn — APPROVED** [review]

File: src/spec/openapi/openapi.ts (hunk 1 above)
Comment (pevisscher) [reply src/spec/openapi/openapi.ts:283]: Most tooling dont support this fully yet, keeping it there for when time progresses to when we can correctly specify our dialect :)

**Comment by pevisscher** [comment]
### Merge activity

* **Dec 3, 11:55 AM GMT+1**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/skyleague/event-horizon/217).

**Pull request merged**